solanaj/src/main/java/org/p2p/solanaj/utils/PublicKeyValidator.kt

```diff
@@ -0,0 +1,22 @@
+package org.p2p.solanaj.utils
+
+import org.bitcoinj.core.Base58
+import org.p2p.solanaj.core.PublicKey
+
+object PublicKeyValidator {
+    private const val REGEX = "[1-9A-HJ-NP-Za-km-z]{32,44}"
+
+    @Throws(IllegalArgumentException::class)
+    fun validate(publicKey: String) = require(
```

Comment:
explicit return type here please

Reply:
oh, it's Unit type here, eh? It was hard to compile the code in my head and guess the return type. I thought it was Boolean really....
Maybe block body here then? It's easy to forget that require is Exception throwing func

Reply:
This `require` function has a `lazyMessage` block param after the check so aesthetically it better not to build the nested brackets construction. That's why I prefer the expression body over the block one

Reply:
nested brackets and indents problems are solved by the "Extract Method" pattern, so I don't think that's a major problem here.

IMHO the problem with the expression body is that most of the time methods are written and read in different way. 
i.e.
I can write this method this way
```kotlin
fun validate(publicKey: String) = require(Regex(REGEX).matches(publicKey) 
        && Base58.decode(publicKey).size == PublicKey.PUBLIC_KEY_LENGTH)
```
or this way
```kotlin
fun validate(publicKey: String) = 
    require(Regex(REGEX).matches(publicKey) && Base58.decode(publicKey).size == PublicKey.PUBLIC_KEY_LENGTH) // kinda better
```

nevertheless, expression body breaks reading rules that we all expect: 
```
method_signature_with_return_type { // where the bracket shows that signature (short public API) finished and now implementation begins
    body/logic/smth
} // impl finishes
 ```

but with block body you are forced to put all this stuff inside where it belongs:
```kotlin
fun validate(publicKey: String) { // now im sure that there is no return value here! never sure about expression body tho
    require(Regex(REGEX).matches(publicKey) 
        && Base58.decode(publicKey).size == PublicKey.PUBLIC_KEY_LENGTH)
}
```

I also can show this in plain text enumeration example:
```
To use this app properly you need to : 1. download app from store // ??? brain is asking why the hell part of the instruction is here
    2. open app 
    3. use it
    4. ???
    5. PROFIT
```

Code is like a book. When you are reading a book, you know that each paragraph split by new lines has its own sense. 
Like method signature. We describe what this method expect and what it returns (first idea), and then on new line you can expect another thing - implementation (second idea)

Reply:
I hope I don't look like a nerd, I'm okay with expression body till they have an explicit return type. But I think we're going to talk about it in our code style meeting, so nvm.

Reply:
Thank you for the research 🙌